pilot/pkg/serviceregistry/serviceentry/store.go

```diff
@@ -63,6 +63,14 @@ func (s *serviceInstancesStore) addInstances(key configKey, instances []*model.S
 		if _, f := s.instances[ikey]; !f {
 			s.instances[ikey] = map[configKey][]*model.ServiceInstance{}
 		}
+		if instance.Service.Resolution == model.DNSRoundRobinLB {
```

Comment:
This is another idea I was saying. Basically check and do not add. We can probably optimize by storing hosts so that we can improve on key lookup. And log a warning here.